electron/modules/sensors.js

```diff
@@ -0,0 +1,68 @@
+const cache = require('../services/cache.service.js')
+const datastore = require('../datastore');
+
+var identifySensor = exports.identifySensor = (userId, sensorId) => {
+    return new Promise (async (resolve, reject) => {
+        try {
+            // 1) Check if in cache
+            let value = await isSensorInCache(userId)
+            if (value) {
+                if (value.sensors[sensorId]) {
+                    return resolve(true)
+                }
+            }
+            // 2) Look for sensor in local database
+            let isNewSensor = await datastore.findUserSensor(userId, sensorId);
+            // 3) If not in database, add to database
+            if (!isNewSensor) {
+                datastore.addUserSensor(userId, sensorId);
+            }
+            // 4) Add to cache 
+            await insertSensorToCache(userId, sensorId)
+            return resolve(false)
+        } catch (error) {
+            console.log(error)
+            return reject(error)
+        }
+        
+    })
+    
+};
+
+var isSensorInCache = exports.isSensorInCache = (userId) => {
+    return new Promise (async (resolve, reject) => {
+        try {
+            let value = await cache.get(userId)
+            if (value) {
+                return resolve(value)
+            }
+            return resolve(false)
```

Comment:
I notice you were using `return reject` when ever a function has to return a false argument. Notice that returning `false` meaning there are no sensors in the cache (in this case), does not mean that there was an error in the execution of the method. `return reject()` will throw an error to the caller, and that is something that will cause undesired results such as `failing test cases` etc. Please use `return reject()` only when an error has occured